### PR TITLE
Enable Org SAML handling

### PR DESCRIPTION
**Pull request by dataclouder**
Addresses Issues #565 and #1041 

Implements 
* data source `vcd_org_saml` to read the SAML settings
* data source `vcd_org_saml_metadata` that reads (and optionally saves on file) the service provider metadata that you can feed to the identity provider.
* resource `vcd_org_saml` that allows the creation and modification of SAML settings. You will need a metadata file retrieved from the identity provider (not the one retrieved from `vcd_org_saml_metadata`) to enable the service.

### CODE REVIEW

File: website/docs/r/org_saml.html.markdown [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+output "result" {
+  value = data.http.example.response_body
+}
+```
+To save into a file, we can run the following:
+
+```
+$ terraform apply -auto-approve
+$ terraform output -raw result > data.xml
```
Comment (Didainius) [inline website/docs/r/org_saml.html.markdown:73]: Not sure if we need that for automated approach, but something like https://registry.terraform.io/providers/hashicorp/local/latest/docs/resources/file could also be used to store the output without jumping to console

File: vcd/resource_vcd_org_saml.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+			"entity_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Your service provider entity ID. Once you set this field, it cannot be changed back to empty.",
+			},
+			"identity_provider_metadata_file": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the file containing the metadata from the identity provider",
+			},
```
Comment (Didainius) [inline vcd/resource_vcd_org_saml.go:44]: My only thought here is - should we also accept text entry. That could be convenient if one uses another datasource to fetch the metadata 

https://github.com/vmware/terraform-provider-vcd/pull/1064/files#diff-6c14c3866e8fd9700b3a55a1aa50c84df75adda9644a846f297f4bc3af8f1b7aR61-R63

File: website/docs/r/org_saml.html.markdown (hunk 1 above)
Comment (dataclouder) [reply website/docs/r/org_saml.html.markdown:73]: Once I added the ability of using metadata from text, I decided to remove this paragraph completely

File: website/docs/r/org_saml.html.markdown [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_org_saml"
+sidebar_current: "docs-vcd-resource-org-saml"
+description: |-
+  Provides a VMware Cloud Director Organization SAML resource. This can be used to create, delete, and update SAML configuration for an organization .
+---
+
+# vcd\_org\_saml
+
+Provides a VMware Cloud Director Org SAML resource. This can be used to create, update, and delete SAML configuration for an organization.
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply website/docs/r/org_saml.html.markdown]: fixed

File: website/docs/r/org_saml.html.markdown [hunk 4]
```diff
@@ -0,0 +1,107 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_org_saml"
+sidebar_current: "docs-vcd-resource-org-saml"
+description: |-
+  Provides a VMware Cloud Director Organization SAML resource. This can be used to create, delete, and update SAML configuration for an organization .
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply website/docs/r/org_saml.html.markdown]: fixed

File: vcd/resource_vcd_org_saml.go [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+	if err != nil {
+		return diag.Errorf("[Org SAML delete] error unsetting SAML settings for Org %s: %s", adminOrg.AdminOrg.Name, err)
+	}
+	return nil
+}
+
+// resourceVcdOrgSamlImport is responsible for importing the resource.
+// The only parameter needed is the Org identifier, which could be either the Org name or its ID
+func resourceVcdOrgSamlImport(_ context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	orgName := d.Id()
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply vcd/resource_vcd_org_saml.go]: changed

File: website/docs/d/org_saml_metadata.html.markdown [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+# The metadata will be stored in vcd-metadata.txt
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org_id` - (Required)  - ID of the organization containing the SAML metadata
```
Comment (adambarreiro) [inline website/docs/d/org_saml_metadata.html.markdown:35]: nitpick
```suggestion
* `org_id` - (Required) ID of the organization containing the SAML metadata
```

**Review by adambarreiro — COMMENTED** [review]
Nice PR, just a few comments/nits

**Review by Didainius — APPROVED** [review]
Thanks for adjustments - my testing worked on 10.4.1 and 10.4.2

File: vcd/datasource_vcd_org_saml_metadata.go [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+	vcdClient := meta.(*VCDClient)
+
+	orgId := d.Get("org_id").(string)
+
+	adminOrg, err := vcdClient.GetAdminOrgById(orgId)
+	if err != nil {
+		return diag.Errorf("[Org SAML metadata retrieval] error searching for Org %s: %s", orgId, err)
+	}
+
+	util.Logger.Printf("--- start retrieving\n")
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply vcd/datasource_vcd_org_saml_metadata.go]: Thanks. It was a leftover. Removed

File: vcd/datasource_vcd_org_saml_metadata_test.go [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+			{
+				PreConfig: func() {
+					if fileExists(metadataFileName) {
+						_ = os.Remove(metadataFileName)
+					}
+				},
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckVcdOrgExists(datasource1),
+					// The retrieved metadata should contain both the VCD URL and the name of the target organization`
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply vcd/datasource_vcd_org_saml_metadata_test.go]: fixed

File: vcd/resource_vcd_org_saml.go [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+		return diag.Errorf("[Org SAML %s] error searching for Org %s: %s", origin, orgId, err)
+	}
+
+	fileName := d.Get("identity_provider_metadata_file").(string)
+	metadataText := d.Get("identity_provider_metadata_text").(string)
+
+	if fileName != "" {
+		metadataFromFile, err := os.ReadFile(fileName) // #nosec G304 -- We need user input for this file
+		if err != nil {
+			return diag.Errorf("[ORG SAML %s %s] error reading metadata file %s: %s", origin, adminOrg.AdminOrg.Name, fileName, err)
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply vcd/resource_vcd_org_saml.go]: fixed

File: website/docs/r/org_saml.html.markdown [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+  role                            = "role"
+  group                           = "group"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org_id` - (Required) Org ID: there is only one SAML configuration available for an organization. Thus, the resource can be identified by the Org
```
Comment (lvirbalas) [inline website/docs/r/org_saml.html.markdown:69]: Could this explanation be rephrased? It reads strangely currently. Also, the "Org ID:" is extra/not needed.

**Review by lvirbalas — COMMENTED** [review]
Great feature! Just a few comments inline.

File: website/docs/r/org_saml.html.markdown [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+  org_id                          = data.vcd_org.my-org.id
+  enabled                         = true
+  identity_provider_metadata_file = "somefile.xml"
+}
+```
+
+You can import such SAML configuration into terraform state using one of the following commands
+
+```
+# EITHER
```
[the remark this answers is not in the document]
Comment (dataclouder) [reply website/docs/r/org_saml.html.markdown]: removed

File: website/docs/r/org_saml.html.markdown (hunk 10 above)
Comment (dataclouder) [reply website/docs/r/org_saml.html.markdown]: rephrased

**Review by lvirbalas — APPROVED** [review]
Woohoo!

**Review by adezxc — APPROVED** [review]
Would it be useful to mention in documentation that the resources/datasources are only available to sysadmins?

**Review by adambarreiro — APPROVED** [review]
LGTM!

**Pull request merged**